### PR TITLE
Environment variable support for mi components

### PR DESCRIPTION
**Pull request by chameerar**
Task - https://github.com/wso2-enterprise/choreo/issues/18973

### CODE REVIEW

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 1]
```diff
@@ -59,7 +59,11 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
 
 You can use environment variables in WSO2 Micro Integrator to improve the configuration management, security, portability, and manageability of the Micro Integrator installation. By using environment variables, organizations can simplify the management and maintenance of their micro installations and ensure that they can be quickly and easily updated when required.
 
-The Choreo DevOps Portal allows you to define configurations and secrets to a deployed integration (as well as other component types). For more information, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).
+The Choreo Console allows you to define enviornment variables for your integration project. You can define the environment variables before you deploy the component.
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:62]:
```suggestion
The Choreo Console allows you to define environment variables for your integration project. You can define the environment variables before you deploy the component.
```

Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: Fixed in fd88d99a57c17c5241a87960c3e13d754445543a

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 2]
```diff
@@ -59,7 +59,11 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
 
 You can use environment variables in WSO2 Micro Integrator to improve the configuration management, security, portability, and manageability of the Micro Integrator installation. By using environment variables, organizations can simplify the management and maintenance of their micro installations and ensure that they can be quickly and easily updated when required.
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:61]:
```suggestion
You can use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.

```

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 3]
```diff
@@ -57,9 +57,13 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
 
 ## Environment variables
 
-You can use environment variables in WSO2 Micro Integrator to improve the configuration management, security, portability, and manageability of the Micro Integrator installation. By using environment variables, organizations can simplify the management and maintenance of their micro installations and ensure that they can be quickly and easily updated when required.
+You can use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:60]:
```suggestion
It is recommended to use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.
```

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 4]
```diff
@@ -57,9 +57,13 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
...
 
-The Choreo DevOps Portal allows you to define configurations and secrets to a deployed integration (as well as other component types). For more information, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).
+The Choreo Console allows you to define environment variables for your integration project. You can define the environment variables before you deploy the component.
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:62]:
```suggestion
The component deploy page offers a user-friendly interface for managing environment variables across different environments. For more advanced use cases, the Choreo DevOps portal is available.

The Choreo DevOps Portal allows for greater flexibility in defining configurations and secrets for deployed integrations, as well as other component types, to meet more complex configuration requirements. Refer to the section on Configurations and secrets for further details.
```

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 5]
```diff
@@ -57,9 +57,13 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
...
+
+![Environment Variables](../../assets/img/ipaas/integration-studio/env_variables.png){.cInlineImage-half}
+
+The Choreo DevOps Portal also allows you to define configurations and secrets to a deployed integration (as well as other component types) with more flexibility to cater your advance configuration needs. For more information, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:66]:
```suggestion
The Choreo DevOps Portal allows for greater flexibility in defining configurations and secrets for deployed integrations, as well as other component types, to meet more complex configuration requirements. Refer to the section on [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets) for further details.
```

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 6]
```diff
@@ -57,9 +57,13 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
...
 
 For a list of parameters that you can configure as environment variables, see [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).
```
Comment (a5anka) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:68]:
```suggestion
For a comprehensive list of parameters that can be configured as environment variables, please refer to the [WSO2 API Manager Documentation - Injecting Parameters - Supported parameters](https://apim.docs.wso2.com/en/latest/integrate/develop/injecting-parameters/#supported-parameters).
```

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 3 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 4 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 5 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 6 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: Fixed in 0a219c3d8424e846c56f75bacb3c16c854cae75d

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 7]
```diff
@@ -57,11 +57,15 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
 
 ## Environment variables
 
-You can use environment variables in WSO2 Micro Integrator to improve the configuration management, security, portability, and manageability of the Micro Integrator installation. By using environment variables, organizations can simplify the management and maintenance of their micro installations and ensure that they can be quickly and easily updated when required.
+It is recommended to use environment variables when developing integration artifacts with WSO2 Integration Studio to improve configuration management, security, portability, and manageability. By using environment variables, organizations can simplify the management and maintenance of their integrations and ensure that they can be quickly and easily updated when required.
```
Comment (shaniR) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:60]: Are you trying to say that we recommend that they use environment variables? or is it as an alternative to another option?

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md [hunk 8]
```diff
@@ -57,11 +57,15 @@ During deployment, Choreo performs a Trivy scan to detect critical vulnerabiliti
...
 
-The Choreo DevOps Portal allows you to define configurations and secrets to a deployed integration (as well as other component types). For more information, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).
+The component deploy page offers a user-friendly interface for managing environment variables across different environments. For more advanced use cases, the Choreo DevOps portal is available.
```
Comment (shaniR) [inline en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:62]: Would you be able to explain what message we need to convey here?

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 7 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:60]: We recommend that they use environment. It is the better option here.

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 8 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:62]: The users can manage environment variables for the general use cases through the component deployment page. If they have more advanced use cases, they can do the configurations in Choreo DevOps portal.

Comment (shaniR) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md:62]: The component's **Deploy** page allows you to easily manage environment variables across environments via a user-friendly interface. For more information on how you can manage environment variables for more advanced use cases, see [Configurations and secrets](../../devops/devops-portal.md#configurations-and-secrets).

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md (hunk 8 above)
Comment (chameerar) [reply en/docs/ipaas/micro-integrator/develop-integrations-with-integration-studio.md]: Fixed in a1a9d6250d9d356ba65d8826e0ee55a40d3a258e

**Review by shaniR — APPROVED** [review]

**Pull request merged**